scripts/alembic_with_settings.py

```
@@ -0,0 +1,105 @@
+        parser.error("Cannot combine --workspace with --config/--env")
+
+    db_path = _resolve_state_db_path(args.workspace, args.config_path, args.env_path)
+
+    child_env = os.environ.copy()
+    child_env["STATE_DB_PATH"] = db_path
+
+    cmd = [sys.executable, "-m", "alembic", "-c", str(alembic_config), *alembic_args]
+    print(f"Resolved STATE_DB_PATH={db_path}")
+    print("Running:", " ".join(cmd))
+
+    result = subprocess.run(cmd, cwd=package_root, env=child_env)
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)